scylla/src/transport/connection.rs

```diff
@@ -1721,6 +1728,112 @@ mod tests {
+        // no matter whether coalescing is enabled or not.
+
+        let uri = std::env::var("SCYLLA_URI").unwrap_or_else(|_| "127.0.0.1:9042".to_string());
+        let addr: SocketAddr = resolve_hostname(&uri).await;
+        let ks = unique_keyspace_name();
+
+        {
+            // Preparation phase
+            let session = SessionBuilder::new()
+                .known_node_addr(addr)
+                .build()
+                .await
+                .unwrap();
+            session.query(format!("CREATE KEYSPACE IF NOT EXISTS {} WITH REPLICATION = {{'class' : 'SimpleStrategy', 'replication_factor' : 1}}", ks.clone()), &[]).await.unwrap();
+            session.use_keyspace(ks.clone(), false).await.unwrap();
+            session
+                .query(
+                    "CREATE TABLE IF NOT EXISTS t (p int primary key, v blob)",
+                    &[],
+                )
+                .await
+                .unwrap();
+        }
+
+        let subtest = |enable_coalescing: bool, ks: String| async move {
+            let (connection, _) = super::open_connection(
+                UntranslatedEndpoint::ContactPoint(ContactPoint {
+                    address: addr,
+                    datacenter: None,
+                }),
+                None,
+                ConnectionConfig {
+                    enable_write_coalescing: enable_coalescing,
+                    ..ConnectionConfig::default()
+                },
+            )
+            .await
+            .unwrap();
+            let connection = Arc::new(connection);
+
+            connection
+                .use_keyspace(&super::VerifiedKeyspaceName::new(ks, false).unwrap())
+                .await
+                .unwrap();
+
+            connection
+                .query(&"TRUNCATE t".into(), (), None)
+                .await
+                .unwrap();
+
+            let mut futs = Vec::new();
+
+            for batch_size in 0..10 {
+                // Each future should issue more and more queries in the first poll
+                let base = (batch_size - 1) * batch_size / 2;
+                let conn = connection.clone();
+                futs.push(tokio::task::spawn(async move {
+                    let futs = (base..base + batch_size).map(|j| {
+                        let q = Query::new("INSERT INTO t (p, v) VALUES (?, ?)");
+                        let conn = conn.clone();
+                        async move {
+                            conn.query(&q, (j, vec![j as u8; j as usize]), None)
+                                .await
+                                .unwrap()
+                        }
+                    });
+                    futures::future::join_all(futs).await;
+                }));
+
+                tokio::task::yield_now().await;
+            }
+
+            futures::future::join_all(futs).await;
+
+            // Check that everything was written properly
+            let range_end = (10 - 1) * 10 / 2;
```

Comment:
It might be clearer IMO if you make `10` some meaningfully named variable instead of a magic constant.

Reply:
Done, extracted to a constant: `NUM_BATCHES`